PowerEditor/src/MISC/Common/FileInterface.h

```diff
@@ -0,0 +1,64 @@
+// This file is part of Notepad++ project
+// Copyright (C)2021 Don HO <don.h@free.fr>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// at your option any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+#pragma once
+
+#include <windows.h>
+#include <tchar.h>
+#include <cstdint>
+
+
+class CFile
+{
+public:
+	enum class Mode
+	{
+		READ,
+		WRITE,
+		APPEND
+	};
+
+	CFile(const char *fname, Mode fmode = Mode::READ);
+	CFile(const wchar_t *fname, Mode fmode = Mode::READ);
+
+	~CFile()
+	{
+		Close();
+	}
+
+	bool IsOpened()
+	{
+		return (_hFile != INVALID_HANDLE_VALUE);
+	}
+
+	void Close();
+
+	int_fast64_t GetSize();
+
+	unsigned long Read(void *rbuf, unsigned long buf_size);
+	bool Write(const void *wbuf, unsigned long buf_size);

```

Comment:
Why not the standard `size_t` for the sizes? I'm really curious — maybe I don't know something…

Reply:
@donho What I meant is that it seems more reasonable for these member functions to have `buf_size` or return type as `size_t` to concentrate the necessary casts or other adjustments in one spot than to spread them all over the code.

Also on a 64 bit system this code will fail to write a buffer over 4GB in size, although it's unlikely to happen, but still…